[FOS-11703] Add high-level documentation (#3)
ready-robotics/ral2-sample-driver: README.md

# READY RAL2 Sample Driver

> This repository contains a sample ral2-driver implementation for ForgeOS and meta-data including its configuration,

Reviewer: ```suggestion
This repository contains a sample ral2-driver implementation for ForgeOS and metadata including its configuration,
```

visualization, and device package details.

## Introduction

> The code in this repository is meant to serve only as a basis for implementing a ForgeOS robot driver functionality. The

Reviewer: Maybe this or just remove functionality with the original?
```suggestion
The code in this repository is meant to serve only as a basis for implementing the functionality of a ForgeOS robot driver. The
```

> ready::SampleDriver class implementation stubs all of functions and emits all of the Qt signals necessary to get the

Reviewer: ```suggestion
ready::SampleDriver class implementation stubs all necessary functions and emits all of the Qt signals necessary to get the
```

corresponding change will need to be made to the "robot-driver-library" attribute found in configuration/Sample.json.

### SampleDriver.cpp, SampleDriver.hpp

These source files contain the stubbed ral2::Driver interface implementation in the class `ready::SampleDriver`. These
files should be renamed to match the vendor driver name and should update namespace names to match the vendor.
Additional files can be added to the `src` tree to contain other implementation details as needed.

Changing the names of these files, namespaces, and classes will require changes in the CMakeLists.txt file and the
DriverLib.cpp file.

### configuration/Sample.json

This source file contains high-level configuration of the robot driver. It contains ral2-service attributes, robot model configuration, robot controller configuration, faults, warnings, and supported robot modes.

This file will need to be updated when adding or removing robot models and robot controllers, updating fault messages,
changing jogging behavior, and modifying high-level ral2-service behaviors. The default values are a reasonable starting
point.

### configurations/robot-models/*.json

> Each of the .json files in the robot-models directory corresponds to settings for a specific robot model. Adding and

Reviewer: ```suggestion
Each of the .json files in the robot-models directory correspond to settings for a specific robot model. Adding and
```

> removing these files will require changes to CMakeLists.txt and the following attributes in "configuration/Sample.json":

Reviewer: ```suggestion
removing these files requires changes to CMakeLists.txt and the following attributes in "configuration/Sample.json":
```

* controller.*.supported-robot-models
* attribute-list - robot attributes - model values

### configurations/controller-models/*/*.json

> Each of the .json files in the controller-models directory tree corresponds to a specific operating mode for a

Reviewer: ```suggestion
Each of the .json files in the controller-models directory tree correspond to a specific operating mode for a
```

> controller. Adding and removing these files/directories will require changes to CMakeLists.txt and the following

Reviewer: ```suggestion
controller. Adding and removing these files/directories requires changes to CMakeLists.txt and the following
```

attributes in the "configuration/Sample.json":

* controller.*.supported-robot-models
* controller.*.safety-configuration
* attribute-list - controller attributes - model values
* attribute-list - controller attributes - name ARRAY

### Manifest.json

> The manifest.json file is used to describe the ForgeOS device package. The following fields will need to be updated:

Reviewer: ```suggestion
The manifest.json file is used to describe the ForgeOS device package. The following fields need to be updated:
```

> * "guid" -- This field is a globally unique string identifying the package in ForgeOS. The value is typically of the form "ready-ral-<robot-vendor>-driver"

Reviewer: We currently don't use ral in this field but probably not a bad idea.

> * "guid" -- This field is a globally unique string identifying the package in ForgeOS. The value is typically of the form "ready-ral-<robot-vendor>-driver"
